### PR TITLE
Only display marriage-abroad payment info when services are shown

### PR DESCRIPTION
**Pull request by chrisroos**
__This supersedes PR #2502. I've made a few changes to both the code and the commit notes. I've also rewritten the PR description to (hopefully) better communicate its purpose.__

Trello card: https://trello.com/c/K2bUzopZ

We only want to display payment information (e.g. "You can pay by cash only.") when we display the fees section (i.e. when we render the `consular_fees_table` partial).

The Trello card also covers the links to consular fees ("You normally have to pay fees for consular services...") but I plan to deal with those in a separate branch.

This branch moves the payment information from the various outcome templates to the marriage_abroad_services.yml file. The outcome templates render the new `_payment_information` partial which uses the data in marriage_abroad_services.yml to determine which payment partial to render. I've only added payment information for the countries that have service data in marriage_abroad_services.yml. This ensures that we only display the information when the fees section is also displayed.

I've moved all the payment information partials to the outcomes/payment_information_partials directory and am storing the name of the partial to render in marriage_abroad_services.yml. I would've preferred to store a list of acceptable payment methods in marriage_abroad_services.yml (rather than the partial names) but I don't think that's currently possible given the nature of the payment information sentences. Hopefully we can make the payment information sentences more consistent in future which should allow us to make this sort of change.

In some cases, the payment information was hardcoded in the template. I've moved this payment information to new partials so that I can then store the name of the partial in the marriage_abroad_services.yml file.

I created and used the temporary `add-payment-partials-to-marriage-abroad-services-data.rb` script to modify the marriage_abroad_services.yml data rather than updating it manually. I hope that having the script in the commits makes it easier to see how I've added data to marriage_abroad_services.yml.

For outcomes that are rendered for multiple countries, I used the `marriage-abroad-countries-leading-to-outcome.rb` script to get a list of those countries before updating the `add-payment-partials-to-marriage-abroad-services-data.rb` script.

This branch has exposed a couple of other issues.

* Outcomes for a number of ceremony countries display services/fees but no payment information.
* Outcomes for a number of ceremony countries were displaying different payment information depending on the type of ceremony. I think (hope) this is an oversight as the mechanism in this branch only allows for payment information to be specified for the country and not at a more granular level.
I've added comments to the Trello card to ask how we address both of these issues.

I plan to add another Trello card to work out whether we can reduce the number of payment information partials in the system.

### CODE REVIEW

File: script/add-payment-partials-to-marriage-abroad-services-data.rb [hunk 1]
```diff
@@ -1,8 +1,124 @@
...
+  tunisia
+  turkmenistan
+  uzbekistan
+  venezuela
+  western-sahara
 )
 
+# Exclude #country_without_consular_facilities? countries
+countries = countries -
+  SmartAnswer::Calculators::MarriageAbroadDataQuery::COUNTRIES_WITHOUT_CONSULAR_FACILITIES
```
Comment (floehopper) [inline script/add-payment-partials-to-marriage-abroad-services-data.rb]: Minor: It wasn't immediately obvious to me why you needed to exclude these countries. It would be nice to explain a bit more either in the comment or in the commit note.

File: script/add-payment-partials-to-marriage-abroad-services-data.rb [hunk 2]
```diff
@@ -1,8 +1,124 @@
...
+
+# Excluded countries ignored in outcome template
+countries = countries -
+  %w(cote-d-ivoire burundi)
```
Comment (floehopper) [inline script/add-payment-partials-to-marriage-abroad-services-data.rb]: Minor: It wasn't immediately obvious to me why you needed to exclude these countries. It would be nice to explain a bit more either in the comment or in the commit note. I assume it's because of this condition `elsif %w(cote-d-ivoire burundi).exclude?(calculator.ceremony_country)`.

File: lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_kosovo_when_residing_in_uk_or_kosovo.govspeak.erb [hunk 3]
```diff
@@ -34,7 +34,8 @@
 
     You normally have to pay fees for consular services in the local currency - these are shown in the list of [consular fees for <%= calculator.country_name_lowercase_prefix %>](/government/publications/<%= calculator.ceremony_country %>-consular-fees).
```
Comment (floehopper) [inline lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_kosovo_when_residing_in_uk_or_kosovo.govspeak.erb]: This line of content seems to include an aspect of how to pay i.e. "in the local currency". Do you think that also needs to be extracted somehow?

File: lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_kosovo_when_residing_in_uk_or_kosovo.govspeak.erb [hunk 4]
```diff
@@ -97,7 +98,8 @@
...
```
Comment (floehopper) [inline lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_kosovo_when_residing_in_uk_or_kosovo.govspeak.erb]: This line of content seems to include an aspect of how to pay i.e. "in the local currency". Do you think that also needs to be extracted somehow?

File: script/add-payment-partials-to-marriage-abroad-services-data.rb [hunk 5]
```diff
@@ -1,9 +1,9 @@
...
-  kuwait
+  laos
 )
 
 payment_method_partial = {
-  'default' => 'pay_by_cash_or_credit_card_no_cheque_nor_amex'
+  'default' => 'pay_by_cash_or_credit_card_no_cheque'
 }
 
 marriage_abroad_services_file = Rails.root.join('lib', 'data', 'marriage_abroad_services.yml')
```
Comment (floehopper) [inline script/add-payment-partials-to-marriage-abroad-services-data.rb]: The commit message talks about Kuwait, but the commit is about Laos.

File: script/add-payment-partials-to-marriage-abroad-services-data.rb [hunk 6]
```diff
@@ -1,14 +1,26 @@
...
+  finland
+  germany
+  iceland
+  luxembourg
+  slovenia
+  sweden
 )
 
+# Exclude countries ignored in the outcome template
+countries = countries - %w(czech-republic cote-d-ivoire)
```
Comment (floehopper) [inline script/add-payment-partials-to-marriage-abroad-services-data.rb]: Minor: I think it would be useful to explain that this was added because of this conditional statement in the outcome template: `elsif %w(czech-republic cote-d-ivoire).exclude?(calculator.ceremony_country)`. Either in the comment or the commit note.

File: lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_kosovo_when_residing_in_uk_or_kosovo.govspeak.erb (hunk 3 above)
Comment (floehopper) [inline lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_kosovo_when_residing_in_uk_or_kosovo.govspeak.erb]: I've just realised this maybe something that you're planning to do in a different PR. Please ignore if that's the case.

File: lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_kosovo_when_residing_in_uk_or_kosovo.govspeak.erb (hunk 4 above)
Comment (floehopper) [inline lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_kosovo_when_residing_in_uk_or_kosovo.govspeak.erb]: I've just realised this maybe something that you're planning to do in a different PR. Please ignore if that's the case.

File: lib/smart_answer/calculators/marriage_abroad_calculator.rb [hunk 7]
```diff
@@ -325,10 +325,14 @@ def services_data_for_country_and_partner_sex_and_residency?
...
     def services_data_for_country_and_partner_sex?
-      @services_data.has_key?(ceremony_country) &&
+      services_data_for_country? &&
         @services_data[ceremony_country].has_key?(@sex_of_your_partner)
     end
 
+    def services_data_for_country?
+      @services_data.has_key?(ceremony_country)
+    end
+
```
Comment (floehopper) [inline lib/smart_answer/calculators/marriage_abroad_calculator.rb]: Minor: Why did you decide to extract `@services_data.has_key?(ceremony_country)` and not `@services_data[ceremony_country]`? If the latter defaulted to returning an empty Hash, I think you could get rid of the `if` statement in `services_payment_partial_name` and first condition in `services_data_for_country_and_partner_sex`. I think getting rid of conditionals is generally a good thing.

**Comment by floehopper** [comment]
I found this much easier to follow than #2502. All my comments/questions are pretty minor and I'd be happy to see this merged as it stands. Good work! 👍

**Comment by chrisroos** [comment]
Thanks for reviewing again, @floehopper.

I'm closing this to rebase on master and address your feedback before opening a new PR to merge.

File: script/add-payment-partials-to-marriage-abroad-services-data.rb (hunk 1 above)
Comment (chrisroos) [inline script/add-payment-partials-to-marriage-abroad-services-data.rb]: I've updated the commit note to explain that the excluded countries are there to match the behaviour in the outcome template. See this commit in PR #2511: https://github.com/alphagov/smart-answers/pull/2511/commits/56479fbe7e1279f1089a91ffb44cd2ed6110980c.

File: script/add-payment-partials-to-marriage-abroad-services-data.rb (hunk 2 above)
Comment (chrisroos) [inline script/add-payment-partials-to-marriage-abroad-services-data.rb]: I've updated the commit note to explain that the excluded countries are there to match the behaviour in the outcome template. See this commit in PR #2511: https://github.com/alphagov/smart-answers/pull/2511/commits/56479fbe7e1279f1089a91ffb44cd2ed6110980c.

File: lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_kosovo_when_residing_in_uk_or_kosovo.govspeak.erb (hunk 3 above)
Comment (chrisroos) [inline lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_kosovo_when_residing_in_uk_or_kosovo.govspeak.erb]: I am planning to address that text separately. I've updated the PR description to make that clearer.

File: lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_kosovo_when_residing_in_uk_or_kosovo.govspeak.erb (hunk 4 above)
Comment (chrisroos) [inline lib/smart_answer_flows/marriage-abroad/outcomes/outcome_opposite_sex_marriage_in_kosovo_when_residing_in_uk_or_kosovo.govspeak.erb]: I am planning to address that text separately. I've updated the PR description to make that clearer.

File: script/add-payment-partials-to-marriage-abroad-services-data.rb (hunk 5 above)
Comment (chrisroos) [inline script/add-payment-partials-to-marriage-abroad-services-data.rb]: I've updated the commit note in PR #2511. See https://github.com/alphagov/smart-answers/pull/2511/commits/e53699ed16b6ae009e4ba7bb6458940b90e202b8.

File: script/add-payment-partials-to-marriage-abroad-services-data.rb (hunk 6 above)
Comment (chrisroos) [inline script/add-payment-partials-to-marriage-abroad-services-data.rb]: I've updated the commit note to explain that the excluded countries are there to match the behaviour in the outcome template. See this same commit in PR #2511: https://github.com/alphagov/smart-answers/pull/2511/commits/1db82d9314a84ad1dce0a8c4e2e3c339e0e33bc7

File: lib/smart_answer/calculators/marriage_abroad_calculator.rb (hunk 7 above)
Comment (chrisroos) [inline lib/smart_answer/calculators/marriage_abroad_calculator.rb]: I think `services_data_for_country?` is currently more consistent with the other similar methods in the `MarriageAbroadCalculator` (see all the [`services_*?` methods in the calculator][calc-services-methods]).

I agree that it can be improved but am happy to leave it for another time.

[calc-services-methods]: https://github.com/alphagov/smart-answers/blob/master/lib/smart_answer/calculators/marriage_abroad_calculator.rb#L294

File: script/add-payment-partials-to-marriage-abroad-services-data.rb (hunk 2 above)
Comment (floehopper) [inline script/add-payment-partials-to-marriage-abroad-services-data.rb]: 👍

File: script/add-payment-partials-to-marriage-abroad-services-data.rb (hunk 1 above)
Comment (floehopper) [inline script/add-payment-partials-to-marriage-abroad-services-data.rb]: 👍

File: lib/smart_answer/calculators/marriage_abroad_calculator.rb (hunk 7 above)
Comment (floehopper) [inline lib/smart_answer/calculators/marriage_abroad_calculator.rb]: 👍
